model/metric.go

```diff
@@ -18,13 +18,25 @@ import (
 	"regexp"
 	"sort"
 	"strings"
+	"unicode/utf8"
+)
+
+type validationSchemeId int
+
+const (
+	LegacyValidation = validationSchemeId(0)
+	UTF8Validation   = validationSchemeId(1)
```

Comment:
Does this work?

```suggestion
	LegacyValidation validationSchemeId = iota
	UTF8Validation
```

Maybe make `validationSchemeId` and call it `ValidationScheme` ? That would be closer to how I understand Go (pseudo) enums.

Reply:
nice -- I couldn't figure out the syntax